docs/design/PT-166642114-miner-activated-consensus.md

```diff
@@ -0,0 +1,146 @@
+# Miner Signalled Consensus Upgrade: Design
+
+This document describes the technical specifications for the miner signalled consensus upgrade whose [functional specifications are available][fspecs].
+
+[fspecs]: https://github.com/aeternity/protocol/blob/1ccb7fb099f7da9ba682b0039d37790050499ed1/consensus/miner_signalled_consensus.md
+
+## Technical specifications
+
+### Behaviour
+
+#### User configuration
+
+The user may specify via the user configuration file the greatest consensus protocol version in such user configuration file to be activated depending on miner signalling.
+At node startup, the node must parse, validate and load such configuration.
+As miner signalling uses block header field `info`, the [current (i.e. old) consensus protocol version C1][fspecs] must be strictly greater than Minerva.
+
+#### Blocks
+
+##### Blocks connected to genesis
+
+The `aec_conductor` process, that is the entry point for all changes to the chain storage, enforces that each block being inserted is connected to genesis.
+By exploiting such current design, the expected consensus protocol version of a block can be computed whenever:
+- An untrusted block is meant to be able to be inserted in the storage.
+  This includes the cases of:
+  - Sync worker process intending to add one block to the storage.
+  - Gossip.
+  - User API.
+- A trusted block is meant to be able to be inserted in the storage.
+  These blocks are not validated.
+  This includes the cases of:
+  - Locally mined or signed block.
```

Comment:
There is also a case when the node runs as a Stratum server and gets a block from a client.

Reply:
[Related refactoring PR](https://github.com/aeternity/aeternity/pull/2624).

Reply:
Addressed in new commit afaa8484d.